adapters/algorix/algorix.go

```diff
@@ -0,0 +1,163 @@
+package algorix
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	EndpointTemplate template.Template
+}
+
+// Builder builds a new instance of the AlgoriX adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	endpoint, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+	bidder := &adapter{
+		EndpointTemplate: *endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+	var errs []error
+
+	adapterRequest, err := a.makeRequest(request)
+	if err == nil {
+		adapterRequests = append(adapterRequests, adapterRequest)
+	} else {
+		errs = append(errs, err)
+	}
+	return adapterRequests, errs
+}
+
+func (a *adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, error) {
+	algorixExt, err := getImpAlgoriXExt(&request.Imp[0])
+
+	if err != nil {
+		return nil, &errortypes.BadInput{Message: "Invalid ExtImpAlgoriX value"}
+	}
+
+	endPoint, err := a.getEndPoint(algorixExt)
+	if err != nil {
+		return nil, err
+	}
+
+	preProcess(request)
+	reqBody, err := json.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     endPoint,
+		Body:    reqBody,
+		Headers: headers,
+	}, nil
+}
+
+// get ImpAlgoriXExt From First Imp. Only check and get first Imp.Ext.Bidder to ExtImpAlgorix
+func getImpAlgoriXExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpAlgorix, error) {
+	var extImpAlgoriX openrtb_ext.ExtImpAlgorix
+	var extBidder adapters.ExtImpBidder
+	err := json.Unmarshal(imp.Ext, &extBidder)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(extBidder.Bidder, &extImpAlgoriX)
+	if err != nil {
+		return nil, err
+	}
+	return &extImpAlgoriX, nil
+}
+
+func (a *adapter) getEndPoint(ext *openrtb_ext.ExtImpAlgorix) (string, error) {
+	endPointParams := macros.EndpointTemplateParams{SourceId: ext.Sid, AccountID: ext.Token}
```

Comment:
Do you think it'd be safer to call `url.PathEscape()` on `ext.Sid` and `ext.Token` just to be sure they are correctly escaped and can safely be placed inside a URL path?
```
92   func (a *adapter) getEndPoint(ext *openrtb_ext.ExtImpAlgorix) (string, error) {
93 -     endPointParams := macros.EndpointTemplateParams{SourceId: ext.Sid, AccountID: ext.Token}
   +     endPointParams := macros.EndpointTemplateParams{
   +         SourceId: url.PathEscape(ext.Sid),
   +         AccountID: url.PathEscape(ext.Token),
   +     }
94       return macros.ResolveMacros(a.EndpointTemplate, endPointParams)
95   }
adapters/algorix/algorix.go
```
Unless you are perfectly confident on this parameter's contents and you deem it unnecessary, of course.